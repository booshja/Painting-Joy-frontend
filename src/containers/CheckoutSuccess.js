// dependencies
import React from "react";
import styled from "styled-components";
// components
import { GoBack, PageTitle } from "../components";
// assets
import YayImg from "../assets/jen-theodore-ualbj7tyJH0-unsplash.jpg";
// breakpoints
import { breakpoints } from "../breakpoints";

const StyledCheckoutCancel = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
`;

const StyledP = styled.p`
    font-family: "News Cycle", sans-serif;
    font-size: 1.2rem;
    text-align: center;
    letter-spacing: 1px;
    width: 80%;
    line-height: 1.2;

    ${breakpoints("font-size", "rem", [{ 414: 1.4 }])}
`;

const StyledImg = styled.img`
    width: 80%;
`;

const CheckoutSuccess = () => {
    return (
        <StyledCheckoutCancel>
            <PageTitle>Congrats!</PageTitle>
            <StyledP>
                It's all yours! You'll receive a receipt from Stripe (our
                payment proccessor) shortly, and we'll send you an email once
                your order has shipped!
            </StyledP>
            <GoBack to={"/"} />
            <StyledImg src={YayImg} alt="Excited child cheering with arms up" />
        </StyledCheckoutCancel>
    );
};

export default CheckoutSuccess;
