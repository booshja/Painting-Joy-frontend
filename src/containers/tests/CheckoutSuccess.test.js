import { render } from "@testing-library/react";
import CheckoutSuccess from "../CheckoutSuccess";
import "jest-styled-components";

/** Smoke Test */
it("renders without crashing", () => {
    render(<CheckoutSuccess />);
});

/** Snapshot Test */
it("matches snapshot", () => {
    const { asFragment } = render(<CheckoutSuccess />);
    expect(asFragment()).toMatchSnapshot();
});

/** RTL Tests */
it("renders correctly", () => {
    const { getByText, getByAltText } = render(<CheckoutSuccess />);

    expect(getByText("Congrats!")).toBeInTheDocument();
    expect(
        getByText(
            "It's all yours! You'll receive a receipt from Stripe (our payment proccessor) shortly, and we'll send you an email once your order has shipped!"
        )
    ).toBeInTheDocument();
    expect(getByText("Back")).toBeInTheDocument();
    expect(
        getByAltText("Excited child cheering with arms up")
    ).toBeInTheDocument();
});
