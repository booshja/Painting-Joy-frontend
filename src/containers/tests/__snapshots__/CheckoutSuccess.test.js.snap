// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  .c3 {
  outline: none;
  background: transparent;
  color: #207070;
  font-family: "News Cycle",sans-serif;
  font-size: 1.4rem;
  font-weight: 700;
  cursor: pointer;
  border: none;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  margin: 0.5rem 0 0.5rem 7.5%;
}

.c3 i {
  margin-right: 0.5rem;
}

.c1 {
  text-align: center;
  font-family: "Barlow Condensed",sans-serif;
  font-size: 2.4rem;
  font-weight: 700;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  text-transform: uppercase;
  padding: 1rem 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  font-family: "News Cycle",sans-serif;
  font-size: 1.2rem;
  text-align: center;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  width: 80%;
  line-height: 1.2;
}

.c4 {
  width: 80%;
}

@media screen and (min-width:414px) {
  .c1 {
    font-size: 2.8rem;
  }
}

@media screen and (min-width:414px) {
  .c2 {
    font-size: 1.4rem;
  }
}

@media screen and (min-width:1024px) {
  .c4 {
    width: 70%;
  }
}

<div
    class="c0"
  >
    <h1
      class="c1"
    >
      Congrats!
    </h1>
    <p
      class="c2"
    >
      It's all yours! You'll receive a receipt from Stripe (our payment proccessor) shortly, and we'll send you an email once your order has shipped!
    </p>
    <button
      class="c3"
    >
      <i
        class="fas fa-long-arrow-alt-left"
      />
      Back
    </button>
    <img
      alt="Excited child cheering with arms up"
      class="c4"
      src="gabe-yay.jpg"
    />
  </div>
</DocumentFragment>
`;
